File: test/fixtures/python/put_with_file.py
import requests

headers = {
    'Content-Type': 'application/x-www-form-urlencoded',
}

with open('new_file') as f:
    data = f.read().replace('\n', '').replace('\r', '').encode()

response = requests.put('http://localhost:28139/upload', headers=headers, data=data)

File: test/fixtures/csharp/put_with_file.cs
using System.Net.Http;
using System.Net.Http.Headers;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Put, "http://localhost:28139/upload");
request.Content = new StringContent(File.ReadAllText("new_file").Replace("\n", string.Empty).Replace("\r", string.Empty));
request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/x-www-form-urlencoded");

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();
